cmake/HPX_SetupBoost.cmake

```diff
@@ -138,7 +140,7 @@ endif()
 
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 link_directories(${Boost_LIBRARY_DIRS})
-if(NOT MSVC)
+if((NOT MSVC) OR HPX_WITH_VCPKG)
```

Comment:
Note quite. VCPKG requires to change several things:

 - explicitly specifying all Boost libraries when linking on Windows (we are relying on auto-linking normally to accommodate for the complex library names in static/dynamic linking scenarios)
 - Always link against the dynamic Boost libraries only as VCPKG provides only those
 - VCPKG add an (undocumented) command line option to MSVC causing hundreds of warnings to be produced (from the Boost.Spirit headers) which I'd like to suppress

I don't expect anybody to ever explicitly use `HPX_WITH_VCPKG` directly, it will be used by the integration of HPX into it only.